core/src/main/scala/cr/pulsar/Consumer.scala

```diff
@@ -210,23 +217,29 @@ object Consumer {
     private case class OptionsImpl[F[_], E](
         initial: SubscriptionInitialPosition,
         logger: E => Topic.URL => F[Unit],
-        unsubscribeMode: UnsubscribeMode
+        unsubscribeMode: UnsubscribeMode,
+        autoAck: Boolean
     ) extends Options[F, E] {
       override def withInitialPosition(
           _initial: SubscriptionInitialPosition
       ): Options[F, E] =
         copy(initial = _initial)
+
       override def withLogger(_logger: E => (Topic.URL => F[Unit])): Options[F, E] =
         copy(logger = _logger)
-      override def withUnsubscribeMode(
-          _unsubscribeMode: UnsubscribeMode
-      ): Options[F, E] =
-        copy(unsubscribeMode = _unsubscribeMode)
+
+      override def withManualUnsubscribe: Options[F, E] =
```

Comment:
Since this is now internal, I think we can get rid of the `UnsubscribeMode` ADT and just use a boolean as we do with `autoAck` no?

Reply:
Yeah, potentially we shouldn't have other modes